### PR TITLE
feat: basic support Apache APISIX 2.10

### PR DESCRIPTION
**Pull request by bzp2010**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Provide basic support for Apache APISIX 2.10. Includes changes to the stream plugins property definition and more.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/conf/schema.json [hunk 1]
```diff
@@ -5204,5 +5345,166 @@
...
+					}
+				},
+				"required": ["protocol_level", "protocol_name", "upstream"],
+				"type": "object"
+			},
+			"version": 0.1
+		}
 	}
-}
+}
```
Comment (starsz) [inline api/conf/schema.json:5510]: Need EOL

Comment (bzp2010) [reply api/conf/schema.json]: fixed

File: web/cypress/fixtures/plugin-dataset.json [hunk 2]
```diff
@@ -463,7 +463,6 @@
         "conn": 1,
         "burst": 0,
         "default_conn_delay": 0.1,
-        "rejected_code": 503,
```
Comment (liuxiran) [inline web/cypress/fixtures/plugin-dataset.json:466]: hi @bzp2010 I noticed that `rejected_code` is still one of the properties in `limit-conn` plugin with default value 503, may I ask why remove it here?

Comment (bzp2010) [reply web/cypress/fixtures/plugin-dataset.json:466]: I think it has default values, so just test the use cases where it has outliers.  check L600

Comment (liuxiran) [reply web/cypress/fixtures/plugin-dataset.json:466]: got it, thanks for your explanation

**Review by liuxiran — APPROVED** [review]

**Comment by liuxiran** [comment]
just wait for ci

**Comment by liuxiran** [comment]
ci ready, @starsz @nic-chen please have a look when you have time, thanks

File: api/test/e2enew/service/service_test.go [hunk 3]
```diff
@@ -614,3 +615,126 @@ var _ = ginkgo.Describe("test service delete", func() {
...
+		"service_id": "s1",
+	}
+	_createRouteBody, err := json.Marshal(createRouteBody)
+	gomega.Expect(err).To(gomega.BeNil())
+
+	table.DescribeTable("test service with hosts",
+		func(tc func() base.HttpTestCase) {
+			base.RunTestCase(tc())
+		},
+		table.Entry("create service without plugin", func() base.HttpTestCase {
```
Comment (nic-chen) [inline api/test/e2enew/service/service_test.go:659]: should have a better desc

Comment (bzp2010) [reply api/test/e2enew/service/service_test.go]: fixed. The desc were updated.

**Review by juzhiyuan — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Pull request merged**